src/DotNet/Writer/ModuleWriterBase.cs

```diff
@@ -384,6 +384,10 @@ public class ModuleWriterOptionsBase {
 				Cor20HeaderOptions.MinorRuntimeVersion = 5;
 			}
 
+			if (module is ModuleDefMD moduleDefMd) {
```

Comment:
There's code below that checks if it's a `ModuleDefMD` so you can move the if block body there instead.